<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Thomas Klammsteiner, PhD">
<meta name="author" content="Sebastian Hupfauf, PhD">

<title>Protocol - Waste Management Lab Course</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="protocol_english_files/libs/clipboard/clipboard.min.js"></script>
<script src="protocol_english_files/libs/quarto-html/quarto.js"></script>
<script src="protocol_english_files/libs/quarto-html/popper.min.js"></script>
<script src="protocol_english_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="protocol_english_files/libs/quarto-html/anchor.min.js"></script>
<link href="protocol_english_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="protocol_english_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="protocol_english_files/libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="protocol_english_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="protocol_english_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="protocol_english_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="protocol_english_files/libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<link rel="stylesheet" href="css/custom.css">
</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-full toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Inhalt</h2>
   
  <ul>
  <li><a href="#schedule" id="toc-schedule" class="nav-link active" data-scroll-target="#schedule">Schedule</a></li>
  <li><a href="#safety-in-the-lab" id="toc-safety-in-the-lab" class="nav-link" data-scroll-target="#safety-in-the-lab">1. Safety in the lab</a>
  <ul class="collapse">
  <li><a href="#laboratory" id="toc-laboratory" class="nav-link" data-scroll-target="#laboratory">1.1. Laboratory</a></li>
  <li><a href="#behaviour-in-the-lab" id="toc-behaviour-in-the-lab" class="nav-link" data-scroll-target="#behaviour-in-the-lab">1.2. Behaviour in the lab</a></li>
  <li><a href="#personal-protective-equipment" id="toc-personal-protective-equipment" class="nav-link" data-scroll-target="#personal-protective-equipment">1.3. Personal protective equipment</a></li>
  <li><a href="#behaviour-in-dangerous-situations" id="toc-behaviour-in-dangerous-situations" class="nav-link" data-scroll-target="#behaviour-in-dangerous-situations">1.4. Behaviour in dangerous situations</a></li>
  </ul></li>
  <li><a href="#introduction" id="toc-introduction" class="nav-link" data-scroll-target="#introduction">2. Introduction</a>
  <ul class="collapse">
  <li><a href="#the-process-of-anaerobic-digestion" id="toc-the-process-of-anaerobic-digestion" class="nav-link" data-scroll-target="#the-process-of-anaerobic-digestion">2.1. The process of anaerobic digestion</a></li>
  </ul></li>
  <li><a href="#material-and-methods" id="toc-material-and-methods" class="nav-link" data-scroll-target="#material-and-methods">3. Material and methods</a>
  <ul class="collapse">
  <li><a href="#experimental-setup" id="toc-experimental-setup" class="nav-link" data-scroll-target="#experimental-setup">3.1. Experimental setup</a>
  <ul class="collapse">
  <li><a href="#reactor-system" id="toc-reactor-system" class="nav-link" data-scroll-target="#reactor-system">3.1.1. Reactor system</a></li>
  <li><a href="#filling-the-reactors" id="toc-filling-the-reactors" class="nav-link" data-scroll-target="#filling-the-reactors">3.1.2. Filling the reactors</a></li>
  <li><a href="#experiment-start" id="toc-experiment-start" class="nav-link" data-scroll-target="#experiment-start">3.1.3. Experiment start</a></li>
  <li><a href="#sampling-and-feeding" id="toc-sampling-and-feeding" class="nav-link" data-scroll-target="#sampling-and-feeding">3.2. Sampling and feeding</a></li>
  </ul></li>
  <li><a href="#gas-analyses" id="toc-gas-analyses" class="nav-link" data-scroll-target="#gas-analyses">3.3. Gas analyses</a>
  <ul class="collapse">
  <li><a href="#pressure-based-measurement-of-gas-volume" id="toc-pressure-based-measurement-of-gas-volume" class="nav-link" data-scroll-target="#pressure-based-measurement-of-gas-volume">3.3.1. Pressure-based measurement of gas volume</a></li>
  <li><a href="#gas-composition" id="toc-gas-composition" class="nav-link" data-scroll-target="#gas-composition">3.3.2. Gas composition</a></li>
  </ul></li>
  <li><a href="#physicochemical-analyses" id="toc-physicochemical-analyses" class="nav-link" data-scroll-target="#physicochemical-analyses">3.4. Physicochemical analyses</a>
  <ul class="collapse">
  <li><a href="#ph-value" id="toc-ph-value" class="nav-link" data-scroll-target="#ph-value">3.4.1. pH value</a></li>
  <li><a href="#electrical-conductivity-ec" id="toc-electrical-conductivity-ec" class="nav-link" data-scroll-target="#electrical-conductivity-ec">3.4.2. Electrical conductivity (EC)</a></li>
  <li><a href="#fostac-ratio" id="toc-fostac-ratio" class="nav-link" data-scroll-target="#fostac-ratio">3.4.3. FOS/TAC ratio</a></li>
  <li><a href="#ammonium-content-nh4" id="toc-ammonium-content-nh4" class="nav-link" data-scroll-target="#ammonium-content-nh4">3.4.4. Ammonium content (NH<sub>4</sub>)</a></li>
  <li><a href="#volatile-fatty-acids-vfas" id="toc-volatile-fatty-acids-vfas" class="nav-link" data-scroll-target="#volatile-fatty-acids-vfas">3.4.5. Volatile fatty acids (VFAs)</a></li>
  <li><a href="#chemical-oxygen-demand-cod" id="toc-chemical-oxygen-demand-cod" class="nav-link" data-scroll-target="#chemical-oxygen-demand-cod">3.4.6. Chemical oxygen demand (COD)</a></li>
  <li><a href="#dry-matter-dm-and-organic-dry-matter-odm" id="toc-dry-matter-dm-and-organic-dry-matter-odm" class="nav-link" data-scroll-target="#dry-matter-dm-and-organic-dry-matter-odm">3.4.7. Dry matter (DM) and organic dry matter (oDM)</a></li>
  <li><a href="#carbonnitrogen-ratio-cn" id="toc-carbonnitrogen-ratio-cn" class="nav-link" data-scroll-target="#carbonnitrogen-ratio-cn">3.4.8. Carbon/Nitrogen ratio (C/N)</a></li>
  </ul></li>
  <li><a href="#hygienization-potential" id="toc-hygienization-potential" class="nav-link" data-scroll-target="#hygienization-potential">3.5. Hygienization potential</a></li>
  <li><a href="#bio-molecular-characterization-of-pathogens" id="toc-bio-molecular-characterization-of-pathogens" class="nav-link" data-scroll-target="#bio-molecular-characterization-of-pathogens">3.6. Bio-molecular characterization of pathogens</a>
  <ul class="collapse">
  <li><a href="#colony-pcr" id="toc-colony-pcr" class="nav-link" data-scroll-target="#colony-pcr">3.6.1. Colony PCR</a></li>
  <li><a href="#dna-purification" id="toc-dna-purification" class="nav-link" data-scroll-target="#dna-purification">3.6.2. DNA purification</a></li>
  <li><a href="#gel-electrophoresis" id="toc-gel-electrophoresis" class="nav-link" data-scroll-target="#gel-electrophoresis">3.6.3. Gel electrophoresis</a></li>
  <li><a href="#dna-quantification" id="toc-dna-quantification" class="nav-link" data-scroll-target="#dna-quantification">3.6.4. DNA quantification</a></li>
  <li><a href="#dna-sequencing-and-blasting" id="toc-dna-sequencing-and-blasting" class="nav-link" data-scroll-target="#dna-sequencing-and-blasting">3.6.5. DNA sequencing and BLASTing</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#bibliography" id="toc-bibliography" class="nav-link" data-scroll-target="#bibliography">4. Bibliography</a></li>
  <li><a href="#appendix" id="toc-appendix" class="nav-link" data-scroll-target="#appendix">Appendix</a></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content column-page-right" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Protocol - Waste Management Lab Course</h1>
<p class="subtitle lead">WT2023/24 | Course No.&nbsp;718837</p>
</div>


<div class="quarto-title-meta-author">
  <div class="quarto-title-meta-heading">Authors</div>
  <div class="quarto-title-meta-heading">Affiliations</div>
  
    <div class="quarto-title-meta-contents">
    <p class="author">Thomas Klammsteiner, PhD <a href="https://orcid.org/0000-0003-1280-5159" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
  </div>
    <div class="quarto-title-meta-contents">
        <p class="affiliation">
            <a href="https://uibk.ac.at">
            Department of Ecology
            </a>
          </p>
      </div>
      <div class="quarto-title-meta-contents">
    <p class="author">Sebastian Hupfauf, PhD <a href="https://orcid.org/0000-0002-2012-9966" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
  </div>
    <div class="quarto-title-meta-contents">
        <p class="affiliation">
            <a href="https://uibk.ac.at">
            Department of Microbiology
            </a>
          </p>
      </div>
    </div>

<div class="quarto-title-meta">

      
  
    
  </div>
  

</header>

<p><br></p>
<p><img src="https://tklammsteiner.github.io/waste-management-course/assets/img/titelbild.png" alt="Title image" width="50%/"></p>
<p><br></p>
<p><strong>Course supervisors</strong>:</p>
<ul>
<li><p>Thomas Klammsteiner, PhD<br>
<a href="mailto:thomas.klammsteiner@uibk.ac.at">thomas.klammsteiner@uibk.ac.at</a></p></li>
<li><p>Sebastian Hupfauf, PhD<br>
<a href="mailto:sebastian.hupfauf@uibk.ac.at">sebastian.hupfauf@uibk.ac.at</a></p></li>
</ul>
<p><br></p>
<p><strong>Teaching assistant</strong>:</p>
<ul>
<li>Laura M. Marte, BSc<br>
<a href="mailto:laura.M.Marte@student.uibk.ac.at">laura.m.marte@student.uibk.ac.at</a></li>
</ul>
<p><br></p>
<p><strong>Source reference cover picture</strong>:</p>
<ul>
<li>Biogas plant Jühnde <a href="http://www.goettinger-tageblatt.de">http://www.goettinger-tageblatt.de</a></li>
<li>Cow in the Alps <a href="https://www.kleinezeitung.at">https://www.kleinezeitung.at</a></li>
</ul>
<p><br></p>
<section id="schedule" class="level1">
<h1>Schedule</h1>
<p><br></p>
<table class="table">
<colgroup>
<col style="width: 23%">
<col style="width: 76%">
</colgroup>
<thead>
<tr class="header">
<th>Abbreviation</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>CNP</td>
<td>Sampling for C/N-Analysis</td>
</tr>
<tr class="even">
<td>CSB</td>
<td>Chemical oxygen demand</td>
</tr>
<tr class="odd">
<td>EC</td>
<td>Electrical Conductivity</td>
</tr>
<tr class="even">
<td>FOS/TAC</td>
<td>Alkalinity (Ratio of organic acids to carbonate buffer capacity)</td>
</tr>
<tr class="odd">
<td>GM</td>
<td>Measuring amount and composition of gas</td>
</tr>
<tr class="even">
<td>PN/PNF</td>
<td>Sampling / Sampling and feeding</td>
</tr>
<tr class="odd">
<td>PT</td>
<td>Pathogen test with selective media</td>
</tr>
<tr class="even">
<td>TS, oTS</td>
<td>Dry matter, organic dry matter</td>
</tr>
<tr class="odd">
<td>VFA</td>
<td>Volatile Fatty Acids</td>
</tr>
<tr class="even">
<td>DNA Q&amp;S</td>
<td>DNA quantification (Quantus) and Sanger sequencing</td>
</tr>
</tbody>
</table>
<p><strong>Objective</strong><br>
The aim of this lab course is to simulate biogas production on a laboratory scale and to learn methods for measuring important monitoring parameters. This involves adjusting various parameters and observing the effects on gas production and other important process parameters.</p>
<p><strong>To bring along</strong><br>
Lab coat, calculator, writing utensils, permanent marker, this lab protocol, lots of motivation</p>
<p><strong>Meeting point</strong><br>
Mo-Fri, 0900AM, Microbiology - practice laboratory (G02), NATWI building, Technikerstraße 25, 2nd floor!</p>
<p><strong>Grading</strong><br>
The final grade for the lab course is calculated in equal parts (25%) from the following four partial grades:</p>
<ul>
<li><p><strong>Introduction test</strong>: On the first day of the lab course, there will be a short introductory test on the content of the lab course protocol. The test will largely consist of multiple-choice questions.</p></li>
<li><p><strong>Collaboration</strong>: The individual co-operation of all students is assessed during the entire internship period.</p></li>
<li><p><strong>Group work</strong>:</p>
<ul>
<li>Each group carries out a literature review on their chosen treatment (e.g.&nbsp;co-digestion, temperature, feeding rate) in parallel with their work in the laboratory. The relevance in practice, the effect on the methane yield, the effect on the microorganisms, unanswered questions and the future potential will be discussed. The results will be presented in groups on the last laboratory day using a PC with a projector (approx. 10 minutes per group).</li>
<li>The most important results of the biogas monitoring should be presented in groups in the form of posters. On the debriefing date, the posters will be presented to the other course participants and the results presented. In addition, results that were not included in the poster will be presented in a short PowerPoint presentation (duration for poster presentation + presentation of further results: approx. 20 minutes per group). The posters do not have to be printed! The presentations will be made using a PC with a projector.</li>
</ul></li>
<li><p><strong>Abstract</strong>: The students prepare an extended abstract (two A4 pages), as is usual for conference participation or scientific publications. This should explain the experimental approach and present the most important results. Further information on writing a scientific abstract will be provided during the lab course. Ideally, it should be submitted by e-mail (to one of the course instructors), with the day of the debriefing set as the deadline.</p></li>
</ul>
<p><br></p>
</section>
<section id="safety-in-the-lab" class="level1">
<h1>1. Safety in the lab</h1>
<section id="laboratory" class="level2">
<h2 class="anchored" data-anchor-id="laboratory">1.1. Laboratory</h2>
<p>Information on the location and use of the laboratory’s safety equipment: first aid equipment, escape routes, telephone, fire alarms, location of fire extinguishing agents, eye and emergency showers, absorbent materials for chemicals, main switch for gas and electricity supply.</p>
</section>
<section id="behaviour-in-the-lab" class="level2">
<h2 class="anchored" data-anchor-id="behaviour-in-the-lab">1.2. Behaviour in the lab</h2>
<ul>
<li>There are no “stupid” questions. If anything is unclear, ask the course instructor or tutor immediately!</li>
<li>Students must never work alone in the lab.</li>
<li>When working with unknown chemicals, everyone must be aware of the hazard symbols on the packaging or consult the Internet for information on safe handling (properties, hazard and safety phrases; Material Safety Data Sheet (MSDS)) and possible sources of danger. Example: Use of the GESTIS database!</li>
<li>Consumption/use and storage of food, drinks, medicines, tobacco products, chewing gum and cosmetics in the laboratory area is prohibited, as is the insertion of contact lenses.</li>
<li>Work surfaces should always be tidy and clean. The “polluter pays” principle applies: everyone cleans and decontaminates the surfaces, containers, equipment or other facilities they use themselves. Spilt material must be disposed of properly immediately!</li>
<li>In the evening, windows and tilt windows must be closed, electrical devices switched off and media connections (compressed air, water, deionised water) switched off.</li>
<li>Hands must be cleaned and disinfected before starting and after completing laboratory work.</li>
<li>All solutions prepared in the lab, containers used, etc. must be labelled with name (or group number), contents and date without exception. Each student is responsible for the correct disposal of materials. This includes</li>
<li>Autoclaving enriched bacterial cultures and contaminated laboratory utensils</li>
<li>Removing the labelling from vessels</li>
<li>Proper cleaning of containers/laboratory utensils</li>
<li>Tidying up the containers/chemical packs in their appropriate storage locations</li>
</ul>
</section>
<section id="personal-protective-equipment" class="level2">
<h2 class="anchored" data-anchor-id="personal-protective-equipment">1.3. Personal protective equipment</h2>
<ul>
<li>It is not permitted to work without a lab coat.</li>
<li>Long trousers and closed shoes must be worn.</li>
<li>Protective gloves must always be worn when handling irritating or toxic substances or media that could contain potential pathogens (e.g.&nbsp;slurry or reactor sludge). Furthermore, safety goggles must always be worn during activities where there is a risk of splashing - we work with pressurised systems! Unknown microorganisms must always be regarded as pathogenic.</li>
<li>Safety goggles must also always be worn when working with acids and bases. (Work with acids and bases may not be carried out in the practical course without a supervisor)</li>
</ul>
</section>
<section id="behaviour-in-dangerous-situations" class="level2">
<h2 class="anchored" data-anchor-id="behaviour-in-dangerous-situations">1.4. Behaviour in dangerous situations</h2>
<ul>
<li>General behaviour: <strong>D</strong>-<strong>A</strong>-<strong>R</strong>-<strong>E</strong> (<strong>D</strong>on’t panic, <strong>A</strong>larm, <strong>R</strong>escue, <strong>E</strong>xtinguish)<br>
</li>
<li>Health and safety violations, near-accidents and any accidents at work must be reported to the lab course supervisors.</li>
</ul>
</section>
</section>
<section id="introduction" class="level1">
<h1>2. Introduction</h1>
<p>By definition, biogas is a combustible, volatile gas produced by the anaerobic fermentation of biomass. The main components are methane (CH<sub>4</sub>, 50-75%) and carbon dioxide (CO<sub>2</sub>, 25-50%). There are also smaller amounts of nitrogen (N<sub>2</sub>), ammonia (NH<sub>3</sub>), hydrogen (H<sub>2</sub>), hydrogen sulphide (H<sub>2</sub>S), water vapour (H<sub>2</sub>O), oxygen (O<sub>2</sub>) and other trace gases. The exact composition depends heavily on the substrate composition and the operating mode of the bioreactor (BMVIT, 2014). The production of biogas is a sustainable and ecological way of generating energy, as no climate-active greenhouse gases are released under optimal conditions. Furthermore, the use of organic residues as a substrate source can be promoted, which simultaneously solves problems associated with waste treatment and disposal. In addition to important ecological effects in the areas of environmental and climate protection, the anaerobic digestion of agricultural waste also offers socio-economic advantages. Another advantage is that the fermentation residues produced, which generally have an improved fertilising effect, a reduced number of pathogens and reduced odour pollution compared to the original substrates, can be applied directly to fields on site (Goberna et al., 2011; Insam et al., 2015).</p>
<p>In agricultural plants, liquid manure (slurry), which is produced during animal husbandry on the farms, is usually used as a basic substrate for biogas production. The use of liquid manure is de facto free of charge, it has a suitable water content and is usually produced in large quantities. However, due to the limited gas formation potential (low C/N ratio), it makes sense to also ferment other, energetically valuable materials as so-called co-substrates (Kaiser, 2007). For example, other agricultural waste products (straw, crop waste, etc.), industrial residues (from bakeries, abattoirs, dairies, breweries, etc.), municipal waste (e.g.&nbsp;shrub cuttings) and collected organic waste from private households can be used for this purpose. In recent years, “NAWARO” plants (renewable raw materials) such as maize, soya, beet or grain, which are planted exclusively for the production of biogas, have also been increasingly used. Although extremely efficient in energy production, the ethical question must be raised as to whether potential foodstuffs should be used for energy production. NAWARO plants are often in direct competition for land with the feed and food industry and the huge monocultures required for this are also questionable from an ecological perspective (EU, 2014). In agricultural plants, liquid manure (slurry), which is produced during animal husbandry on the farms, is usually used as a basic substrate for biogas production. The use of liquid manure is de facto free of charge, it has a suitable water content and is usually produced in large quantities. However, due to the limited gas formation potential (low C/N ratio), it makes sense to also ferment other, energetically valuable materials as so-called co-substrates (Kaiser, 2007). For example, other agricultural waste products (straw, crop waste, etc.), industrial residues (from bakeries, abattoirs, dairies, breweries, etc.), municipal waste (e.g.&nbsp;shrub cuttings) and collected organic waste from private households can be used for this purpose. In recent years, “NAWARO” plants (renewable raw materials) such as maize, soya, beet or grain, which are planted exclusively for the production of biogas, have also been increasingly used. Although extremely efficient in energy production, the ethical question must be raised as to whether potential foodstuffs should be used for energy production. NAWARO plants are often in direct competition for land with the feed and food industry and the huge monocultures required for this are also questionable from an ecological perspective (EU, 2014).</p>
<p>Apart from the substrate (mixture) used, the biogas production process also depends on a number of other parameters, such as temperature, substrate mixing, feed quantity and pH value. The choice of temperature in particular is decisive for the efficiency of the biogas plant and enables operators to control the process. Nowadays, large-scale plants are usually operated under mesophilic (30-40 °C) or thermophilic (50-60 °C) conditions (Jain et al., 2015). However, recent studies show that the gap between these two temperature ranges is particularly interesting, as the advantages of both ranges can be combined here and optimal biogas yields can be achieved (Hupfauf et al., 2018; Tian et al., 2018). The advantages of mesophilic biogas production are high process stability (Labatut et al., 2014) and the low energy required to maintain a constant temperature in the bioreactor. Thermophilic fermentation, on the other hand, leads to better hygienisation of the digestate produced (Sahlström, 2003) and allows higher metabolic rates of the microorganisms involved.</p>
<p>The aim of this lab course is to demonstrate the possible microbial resource utilisation of materials generally declared as “waste”. This is to be illustrated in the form of an anaerobic fermentation experiment with an agricultural residue (cattle manure). In order to demonstrate the influence of various process parameters, the students should decide in groups which adjustments they want to make during the experiment and then observe their effects. In addition to comparing the measured biogas yield and composition, the aim is to familiarise students with the basic parameters for monitoring a biogas plant. In addition, the hygienisation potential will be assessed with the help of a plating experiment on selective media. Selected pure cultures will also be identified using molecular biology.</p>
<section id="the-process-of-anaerobic-digestion" class="level2">
<h2 class="anchored" data-anchor-id="the-process-of-anaerobic-digestion">2.1. The process of anaerobic digestion</h2>
<p>In general, the process of anaerobic digestion can be divided into four main steps (see <b>Fig. 1</b>). During the initial hydrolysis, which determines the speed of the entire process, especially for substrates that are difficult to degrade (Hasegawa et al., 2000), organic polymers are broken down into smaller components (sugars, amino acids, fatty acids, etc.). In addition to obligate anaerobic microorganisms, facultative anaerobic species are also active at this stage. Acidogenic microorganisms subsequently form carbon dioxide (CO<sub>2</sub>), hydrogen (H<sub>2</sub>), ammonium (NH<sub>4</sub>) and organic acids (e.g.&nbsp;propionic, butyric or valeric acid) from these molecules. In the course of the third phase, the acetogenic phase, secondary fermentative bacteria form acetate, H<sub>2</sub> and CO<sub>2</sub>.</p>
<p>These endergonic reactions would not take place under standard conditions, or only slowly, but the local proximity of syntrophic microorganisms and the immediate conversion of the resulting products by methane-forming archaea enable this biochemical metabolic pathway (Conrad et al., 1986).</p>
<p>In methanogenesis, a fundamental distinction can be made between three different pathways (see <b>Tab. 1</b>). In hydrogenotrophic methane formation, CH<sub>4</sub> is formed by the reduction of CO<sub>2</sub> with H<sub>2</sub>. In acetoclastic methane formation, which is quantitatively the most important reaction pathway, CH<sub>4</sub> and CO<sub>2</sub> are obtained from the cleavage of acetate (Ahring et al., 2003). In a third possible metabolic pathway, methyl compounds (C1 compounds) are used. In addition, so-called homoacetogenic bacteria can also form new acetates, which is referred to as homoacetogenesis (Boe, 2006). Conversely, acetate can be converted into CO<sub>2</sub> and H<sub>2</sub> by SAOB (syntrophic acetate-oxidising bacteria) through syntrophic acetate oxidation (SAO). The resulting products are then frequently utilised by syntrophically associated hydrogenotrophic methanogens to form CH<sub>4</sub>. This results in an interspecies transfer of H<sub>2</sub> or CHO<sub>2</sub>-, known as IHT (interspecies hydrogen transfer). Which metabolic pathway is active in a biogas plant always depends on the process conditions, above all the temperature.</p>

<b>Tab. 1:</b> <i>Selection of important metabolic pathways of methanogenesis and their thermodynamic potential.</i>

<table class="table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th>Reaction formula</th>
<th>ΔG0 [kJ mol<sup>-1</sup> CH<sub>4</sub>]</th>
<th>Representative</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>CO<sub>2</sub> + 4 H<sub>2</sub> –&gt; CH<sub>4</sub> + 2 H<sub>2</sub>O</td>
<td>-135</td>
<td>nearly all methanogens</td>
</tr>
<tr class="even">
<td>4 HCOOH –&gt; CH<sub>4</sub> + 3 CO<sub>2</sub> + 2 H<sub>2</sub>O</td>
<td>-130</td>
<td>nearly all methanogens</td>
</tr>
<tr class="odd">
<td>4 CO + 2 H<sub>2</sub>O –&gt; CH<sub>4</sub> + 3 CO<sub>2</sub></td>
<td>-196</td>
<td>Methanothermobacter, Methanosarcina</td>
</tr>
<tr class="even">
<td>4 CH<sub>3</sub>OH –&gt;3 CH<sub>4</sub> + CO<sub>2</sub> + 2 H<sub>2</sub>O</td>
<td>-105</td>
<td>Methanosarcina, methylotrophic methanogens</td>
</tr>
<tr class="odd">
<td>CH<sub>3</sub>OH + H<sub>2</sub> –&gt; CH<sub>4</sub> + H<sub>2</sub>O</td>
<td>-113</td>
<td>Methanomicrococcus, Methanosphaera</td>
</tr>
<tr class="even">
<td>CH<sub>3</sub>COOH –&gt; CH<sub>4</sub> + CO<sub>2</sub></td>
<td>-33</td>
<td>Methanosarcina, Methanosaeta</td>
</tr>
</tbody>
</table>
</section>
</section>
<section id="material-and-methods" class="level1">
<h1>3. Material and methods</h1>
<section id="experimental-setup" class="level2">
<h2 class="anchored" data-anchor-id="experimental-setup">3.1. Experimental setup</h2>
<section id="reactor-system" class="level3">
<h3 class="anchored" data-anchor-id="reactor-system">3.1.1. Reactor system</h3>
<p>Schott bottle reactors have proven themselves for years as robust and easy-to-operate reactor systems on a small laboratory scale. They can be operated either as batch systems or semi-continuous reactors and are therefore extremely flexible in their mode of operation. The main disadvantages are the lack of integrated gas measurement or storage systems, which means that the reactors have to be checked daily, including gas sampling, to prevent the pressure in the reactor system from becoming too high. In addition, there is always a risk of leaks in such a pressurised system, which can lead to significant errors in gas measurement. In addition, increased safety precautions are always required when working with reactors of this type, for example laboratory goggles must always be worn.</p>
<p>We use modified laboratory bottles (2.35 litre volume) with three openings at the top (<b>Fig. 2</b>). These openings allow the connection of various valves, taps or measuring devices and ensure a versatile and customisable use of the reactors. Only one opening (on the side at the top of the reactor) is required for the practical course; the second and third openings are sealed airtight with appropriate sealing caps and suitable sealing rings. A gas-tight brass ball valve is screwed onto the remaining opening. The ball valve is used for feeding, sampling and gas measurement of the reactors. Silicone sealing rings and Teflon sealing tape are used during assembly to ensure complete tightness. To reduce the risk of flying glass splinters in the event of glass breakage, the reactors are encased in a safety net.</p>
</section>
<section id="filling-the-reactors" class="level3">
<h3 class="anchored" data-anchor-id="filling-the-reactors">3.1.2. Filling the reactors</h3>
<p>The reactors are filled together on the first day of the practical course (1 December 2023). The subsequent incubation of the reactors over the weekend before the actual start of the experiment on Monday (d0) enables the system and the microorganisms involved to adapt to the changed conditions and allows the stable operation (without start-up phase) to be investigated during the practical course. For filling, the upper blue lid is unscrewed and 1 litre of sieved cattle slurry (&lt; 2 mm) is filled in using a measuring cylinder. The slurry must be stirred thoroughly beforehand to ensure a uniform substrate in all reactors. The reactor is then tightly closed again and checked for leaks. To do this, a slight overpressure (approx. 0.5 bar) is generated via the ball valve using a bicycle pump (1-2 pump strokes) and the reactor is kept under water. If air bubbles are recognisable, the corresponding point must be better sealed or tightened. The sealed reactors are then incubated at 37 °C with the ball valve closed. Over the weekend, the reactors are shaken daily for 10 seconds and the biogas produced is drained off.</p>
</section>
<section id="experiment-start" class="level3">
<h3 class="anchored" data-anchor-id="experiment-start">3.1.3. Experiment start</h3>
<p>On the first day of the trial (d0), the reactors are prepared by the respective groups according to their chosen control parameter. The following adjustment parameters can be changed (each group decides on one parameter):</p>
<ul>
<li><strong>Feeding quantity</strong>: 50 – 200 mL<br>
</li>
<li><strong>Temperature</strong>: 10 – 60 °C<br>
</li>
<li><strong>Co-substrate</strong>: Waste oil and larvae (<em>Hermetia illucens</em>) Residual substrate<br>
</li>
<li><strong>Co-substrate (waste oil) concentration</strong>: 1 – 20 g L<sup>-1</sup><br>
</li>
<li><strong>Shaking</strong>: 10 – 150 rpm</li>
</ul>
<p><strong>IMPORTANT</strong>: In addition to their four reactors (2 values of the control parameter x 2 replicates), each group also maintains two control reactors. These reactors are operated with the following reference parameters for the entire duration of the experiment: 37 °C, 100 mL feed (without co-substrate), 0 rpm. An average value from both control reactors then serves as a reference for the analysed parameters of the respective group.</p>
</section>
<section id="sampling-and-feeding" class="level3">
<h3 class="anchored" data-anchor-id="sampling-and-feeding">3.2. Sampling and feeding</h3>
<p><strong>ATTENTION</strong>: Sampling and feeding is always carried out after the gas measurements (3.3.1, 3.3.2), as the reactors are pressurised up to this point. Failure to observe this may result in injury and unintentional spraying of the reactor contents!</p>
<p><strong>Durchführung:</strong><br>
To take a sample, a clean 100 mL plastic syringe is attached to the silicone tube of the ball valve. The reactor is then tilted until the liquid phase moves to the top of the Schott bottle. The headspace (gas space) is located at the bottom. The valve is opened and the plastic syringe is slowly drawn up to the 100 ml mark. Close the valve and remove the syringe. Only then may the reactor be set up vertically again! If a blockage occurs during sampling, the opening of the syringe and the ball valve (reactor in upright position!) must be cleaned with a pointed object. The sample taken is transferred to three labelled 2 mL Eppendorf tubes (stored at -20 °C) and a labelled Erlenmeyer flask and is now ready for subsequent analyses. For feeding, a clean 100 mL plastic syringe is filled with 100 mL substrate (mixture) (stir well beforehand!). This is then connected to the ball valve and emptied into the reactor after opening the valve. The reactor does not have to be turned upside down! The ball valve is then completely closed again, the reactor is shaken horizontally for 10 seconds and placed back at its incubation location. The same syringe can be used for feeding all reactors, as long as the substrate mixture is uniform! As small quantities of substrate or reactor sludge can easily escape during sampling and feeding, always work with plenty of cellulose and then clean the workplace with ethanol (70%). <strong>IMPORTANT</strong>: If a group has selected the feeding quantity parameter, other quantities are fed or sampled. However, the amount of material removed must always be replenished with new substrate afterwards! Excess reactor sludge removed is collected and then disposed of. <strong>IMPORTANT</strong>: If a group has selected the co-substrate or co-substrate concentration parameter, feeding is always carried out with a substrate mixture!</p>
</section>
</section>
<section id="gas-analyses" class="level2">
<h2 class="anchored" data-anchor-id="gas-analyses">3.3. Gas analyses</h2>
<p>Determining the amount of gas is of particular interest in large-scale plants, as the biogas produced should bring the desired financial profit. For this purpose, there is generally a whole range of measuring devices that can measure the gas pressure in a certain volume (manometer), the volume displacement (eudiometer) or the flow rate (flowmeter) of the gas. All of these principles can also be applied on a laboratory scale, but in the practical course we limit ourselves to the pressure-based measurement of the gas volume using a manometer.</p>
<section id="pressure-based-measurement-of-gas-volume" class="level3">
<h3 class="anchored" data-anchor-id="pressure-based-measurement-of-gas-volume">3.3.1. Pressure-based measurement of gas volume</h3>
<p>The amount of gas produced each day is determined on the basis of pressure using a digital manometer. Procedure: The reactors are removed individually from the incubators and placed in a well-ventilated area (e.g.&nbsp;near the tilted window). The time the reactor spends outside the incubator must be minimised. Otherwise the process can be significantly disturbed and reproducible measurements are no longer possible. The connecting piece of the manometer is connected to the silicone tube at the end of the ball valve. The ball valve is then opened slowly and the differential pressure displayed between inside and outside the reactor is noted. It should be noted that the pressure gauge needs a certain amount of time to reach a constant value. If the pressure drops continuously, the tightness of the connection must be checked! <strong>ATTENTION</strong>: We are working with pressurised reactor systems here! This requires considered and calm action. It is essential to wear safety goggles during the pressure measurement!</p>
<p><strong>Evaluation</strong>:<br>
The amount of biogas produced is then calculated using the following formula:</p>
<p><span class="math inline">\(Biogas\ [NmL] = \displaystyle \frac{∆P \ [hPa] \ × \ headspace \ [mL] \ × \ DBF(T)}{1013 \ hPa}\)</span></p>
<p>The headspace volume of the reactors in the practical course is 1350 mL, the “dry biogas factor” (DBF, Richards et al.&nbsp;(1991)) for the corresponding temperature is taken from <b>Tab. 2</b>. The result is the amount of gas produced under standard conditions (0 °C, 1013 hPa), taking into account the water vapour contained (= dry biogas). Finally, the change over time of all analysed variants compared to the control as well as the accumulated gas production at the end of the test period are shown graphically (mean values ± standard deviation).</p>

<b>Tab. 2</b>: <i>Temperature-specific “dry biogas factor” (DBF, Richards et al.&nbsp;(1991)) for the calculation of dry biogas.</i>

<table class="table">
<thead>
<tr class="header">
<th></th>
<th>10 °C</th>
<th>20 °C</th>
<th>37 °C</th>
<th>45 °C</th>
<th>55 °C</th>
<th>60 °C</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>DBF</td>
<td>0.953</td>
<td>0.909</td>
<td>0.824</td>
<td>0.773</td>
<td>0.697</td>
<td>0.658</td>
</tr>
</tbody>
</table>
</section>
<section id="gas-composition" class="level3">
<h3 class="anchored" data-anchor-id="gas-composition">3.3.2. Gas composition</h3>
<p>The gas composition is measured using a mobile infrared gas analyser (IRGA; GA5000, Geotech, UK; <b>Fig. 3</b>). This device measures CO<sub>2</sub>, CH<sub>4</sub>, H<sub>2</sub>S, O<sub>2</sub> and residual gas photometrically depending on the adsorption of the emitted infrared light by the gas sample. Each gas sample absorbs different amounts of radiation depending on the gases. Each IRGA can only measure the gases for which reference chambers have been installed. Other measuring methods such as gas chromatography or chemical processes must therefore be used to measure any other trace gases relevant to the process (e.g.&nbsp;H<sub>2</sub>, NH<sub>3</sub>). Regular calibration of the IRGA is important, which is carried out with a calibration gas and ambient air and must be performed every six months for the device. To demonstrate an alternative measurement method, H<sub>2</sub>S measurement using Dräger-Tubes will be demonstrated during the practical course (detailed instructions will be provided during the practical course). The H<sub>2</sub>S content in biogas is of great importance, as this gas causes corrosion damage to metal components (e.g.&nbsp;in combustion engines) in high concentrations (&gt; 200 ppm). Highly contaminated gas must therefore be purified before use. In addition, increased H<sub>2</sub>S concentrations in the anaerobic system are toxic to microorganisms, especially methanogens (at pH 7.8-8: 50% inhibition from 90 mg L<sup>-1</sup>; Koster et al.&nbsp;(1986)), and are also harmful to humans in the air we breathe (irritation from 100-200 ppm!).</p>
<p><strong>Execution</strong>:<br>
For the measurement, 100 mL of biogas is removed from the reactor using a glass syringe. Like the pressure measurement, the sample is taken via the ball valve. The tip of the glass syringe is connected to the silicone tube and the valve is then carefully opened. The gas withdrawal can be precisely controlled by turning the ball valve slightly. Avoid opening the valve too quickly, as otherwise the plunger could be pushed out of the syringe and damage to the material or injuries could occur. The filled glass syringe is then connected to the IRGA and the measurement is started by pressing the PURGE button. Once the syringe is completely empty, the measurement must be stopped manually by pressing the FLUSH button again. After a short equilibration time, the concentrations of the gases CH<sub>4</sub>, CO<sub>2</sub>, H<sub>2</sub>S and O<sub>2</sub> are displayed. The device must be purged with room air between each measurement (until CO<sub>2</sub> and CH<sub>4</sub> &lt; 0.2%)! Immediately after the analysis, the excess biogas is drained into the reactors, thus reducing the overpressure. Discharge takes place exclusively in the designated area (fume cupboard)! <strong>ATTENTION</strong>: As the reactors are still pressurised up to this point, the determination of the gas composition must also be carried out carefully and cautiously. This includes in particular the wearing of protective goggles!</p>
<p><strong>Evaluation</strong>:<br>
The change over time of all biogas components (CH<sub>4</sub>, CO<sub>2</sub>, H<sub>2</sub>S, O<sub>2</sub>) is to be graphically represented. In addition, the methane yield (mL CH<sub>4</sub> g<sup>-1</sup> oTS) of the analysed treatments and the control should be calculated with the help of the gas composition. The choice of a suitable visualisation option is up to the respective groups.</p>
</section>
</section>
<section id="physicochemical-analyses" class="level2">
<h2 class="anchored" data-anchor-id="physicochemical-analyses">3.4. Physicochemical analyses</h2>
<section id="ph-value" class="level3">
<h3 class="anchored" data-anchor-id="ph-value">3.4.1. pH value</h3>
<p>An important feature for the characterization of a bioreactor is the acidity (pH value). The availability of nutrients in particular, but also the toxicity of certain substances (H<sub>2</sub>S, NH<sub>3</sub>) depends on the pH value. Most pH meters are potentiometric devices consisting of a glass sphere filled with buffer solution. Protons are deposited in a thin layer on the surface of the glass sphere, causing a galvanic voltage to build up depending on the pH difference between the inside and outside. This voltage is measured using two electrodes, the measuring electrode inside the glass sphere and a reference electrode outside. Procedure: The pH meter must always be calibrated before the first measurement (according to the instructions provided in the lab)! The sludge samples are then measured in the Erlenmeyer flasks. As soon as the pH meter shows a constant value (this is also indicated by a corresponding symbol on the device), the value is noted. The glass electrode must always be rinsed with A.d. between samples. After the analysis, the electrode is thoroughly rinsed again, carefully dried with cellulose and immersed in the storage solution (KCl). As the device used in the practical course is a temperature-corrected pH meter, the samples do not have to be measured at a uniform temperature. However, large deviations from the air temperature will result in (considerably) longer measurement times. <strong>ATTENTION</strong>: As an acid is required to calibrate the pH meter, protective goggles and gloves must be worn!</p>
<p><strong>Evaluation</strong>:<br>
The course of the pH value should be shown graphically over time for all preparations in comparison to the control (mean ± standard deviation).</p>
</section>
<section id="electrical-conductivity-ec" class="level3">
<h3 class="anchored" data-anchor-id="electrical-conductivity-ec">3.4.2. Electrical conductivity (EC)</h3>
<p>Electrical conductivity (= conductivity) indicates how well a substance can conduct electrical current. In a liquid, the conductivity of a substance depends on the number of dissolved ions. The electrical conductivity therefore correlates strongly with the ammonium concentration, but there are also many other salts in the reactor sludge. Excessive electrical conductivity, and therefore too many ions in the solution, can disrupt the anaerobic fermentation process due to unfavorable osmotic conditions (Ma et al., 2018). The EC is measured in the practical course using a conductivity electrode, analogous to the measurement of the pH value. Procedure: The samples are measured either before or after the pH measurement (3.4.1) in the same Erlenmeyer flask. For this purpose, the EC electrode is immersed in the undiluted reactor sludge and the measured value is noted on the device after settling (constant value for approx. 5 s). The Erlenmeyer flasks must always be swirled before the measurement to ensure a well-mixed, homogeneous sample.</p>
<p><strong>Evaluation</strong>:<br>
The course of the electrical conductivity should be shown graphically over time for all approaches in comparison to the control (mean value ± standard deviation).</p>
</section>
<section id="fostac-ratio" class="level3">
<h3 class="anchored" data-anchor-id="fostac-ratio">3.4.3. FOS/TAC ratio</h3>
<p>The FOS/TAC value describes the ratio of volatile organic acids (FOS) to carbonate alkalinity (total inorganic carbonate, TAC). The alkalinity in the reactor sludge is a measure of the alkaline buffer capacity of the system and determines how robust the pH value is against an accumulation of acids. If a bioreactor has a high alkalinity, a slight accumulation of VFAs will not immediately disrupt the process. FOS/TAC is therefore a standard parameter for assessing the stability of a bioreactor. The measurement is typically carried out by titration to the two pH values 4.4 (FOS) and 5.0 (TAC). In the course of the practical course, titration is carried out using an automatic titrator (<b>Fig. 4</b>) and 0.1 N sulphuric acid.</p>
<p><strong>Execution</strong>:<br>
A small amount of sample (e.g.&nbsp;1 mL) is filled up to 50 mL in a beaker with A.d. and provided with a small magnetic stirrer. The beaker is then placed on the automatic titrator and the pH probe and the acid dosing tube are immersed in the liquid. The glass electrode must never be hit by the magnetic stirrer, as this can easily damage it! The titrator is started and the stirrer is switched on at medium speed. The appliance is controlled via the connected remote control:</p>
<ul>
<li>Press <code>&lt;User Meth&gt;</code> until Load method appears on the display<br>
</li>
<li><code>&lt;Enter&gt;</code><br>
</li>
<li>Press <code>&lt;Select&gt;</code> until FOSTAC appears on the display<br>
</li>
<li><code>&lt;Enter&gt;</code><br>
</li>
<li><code>&lt;Start&gt;</code></li>
</ul>
<p>At the end of the measurement, the respective volumes of acid required to reach pH 5 or 4.4 are displayed. Press <code>&lt;Enter&gt;</code> to switch back and forth between the two points. <strong>IMPORTANT</strong>: The pH electrode must be recalibrated before the first measurement of each group. Calibration is also carried out using the titrator remote control:</p>
<ul>
<li>Press <code>&lt;Mode&gt;</code> until CAL appears on the display<br>
</li>
<li><code>&lt;Enter&gt;</code><br>
</li>
<li><code>&lt;Start&gt;</code><br>
The electrode is immersed in buffer 1 (pH 7)<br>
</li>
<li><code>&lt;Enter&gt;</code><br>
When prompted on the display, the electrode is immersed in buffer 2 (pH 4)<br>
</li>
<li><code>&lt;Enter&gt;</code></li>
</ul>
<p>At the end of the process, the calibration parameters are displayed and the first measurement can be started directly according to the procedure explained above.</p>
<p><strong>ATTENTION</strong>:<br>
As acids are used for both the calibration of the pH electrode and the titration, protective goggles and gloves must be worn!</p>
<p><strong>Evaluation</strong>:<br>
The FOS/TAC ratio is then calculated from the measured values (acid volume) using the following formula:</p>
<p><span class="math inline">\(FOS⁄TAC \ [\frac{mg/L}{mg/L}] \ = \ \frac{(B \ × \ \frac{20}{V_P} \ × \ 1.66 \ - \ 0.15) \ × \ 500 \ × \ V_P}{A \ × \ 5000}\)</span></p>
<p>A = Volume [mL] of the acid needed to reach pH 5.0<br>
B = Volume [mL] of the acid from pH 5.0 to pH 4.4<br>
VP = Volume [mL] of the sample.</p>
<p>The FOS/TAC value should be shown graphically over time for all approaches compared to the control (mean ± standard deviation).</p>
</section>
<section id="ammonium-content-nh4" class="level3">
<h3 class="anchored" data-anchor-id="ammonium-content-nh4">3.4.4. Ammonium content (NH<sub>4</sub>)</h3>
<p>The ammonium content in bioreactors is an important control parameter for monitoring the process. High concentrations, especially under thermophilic conditions, can lead to a considerable process disturbance and an associated drop in gas production. The measurement in the practical course is carried out by means of flow injection analysis (FIA) using the OPA/NAC method (Meseguer-Lloret et al., 2002). The ammonium reacts with ortho-phthaldialdehyde (OPA) and N-acetyl-cysteine (NAC) under alkaline conditions and, after excitation at 415 nm, is measured on a high-performance liquid chromatography (HPLC) system (<b>Fig. 5</b>). Alternatively, ammonium can also be measured classically using a photometer (e.g.&nbsp;Berthelot reaction) or with the aid of rapid cuvette tests (Nanocolor).</p>
<p><strong>Execution</strong>:<br>
The thawed samples are centrifuged at 20,000 g for 10 min. The supernatant is then diluted 1:10 with A.d. and sterile-filtered in short Eprouvettes (pore size: 0.2 µm, filter material: regenerated cellulose RC). A 1:5 dilution with A.d. is then prepared and 250 µL pipetted into a 96-well plate (one common plate for all groups, note the positions!). The remaining filtrate must be stored (and sealed at 4 °C), as it is required for the VFA measurement (3.4.5). As soon as a group has pipetted all samples onto the plate, it must be covered immediately and placed at 4 °C. Otherwise there is a risk of the volatile components escaping quickly! As soon as all groups are ready, the plate is sealed with aluminum foil, taken to the analytical laboratory and loaded into the HPLC device.</p>
<p><strong>Evaluation</strong>:<br>
After the analysis, we receive a PDF data sheet with NH<sub>4</sub>-N concentrations in mg L<sup>-1</sup>. The change in ammonium concentration over time (mean value ± standard deviation) should then be displayed graphically.</p>
</section>
<section id="volatile-fatty-acids-vfas" class="level3">
<h3 class="anchored" data-anchor-id="volatile-fatty-acids-vfas">3.4.5. Volatile fatty acids (VFAs)</h3>
<p>Volatile fatty acids (formate, acetate, propionate, butyrate, valerate) are important intermediate products of anaerobic fermentation and enable the processes within a bioreactor to be monitored precisely. Accumulation generally indicates excessive feeding or an imbalance in the individual phases of the process. In addition, an accumulation of propionate can lead to inhibition of gas production, for example. For this reason, the ratio of acetate to propionate is often used as an indicator of process stability (Weiland, 2008). An accumulation of butyrate or valerate, especially their iso-forms, is another clear indication of stability problems (Drosg, 2013). The volatile fatty acids are measured in the lab using HPLC (<b>Fig. 5</b>). The similar polarity of the compounds in question plays a central role, as this property is used to bind substances to a stationary phase (separation column). The individual substances are then washed from the separation column by changing the polarity of the solvent (eluent) and collected separately. The respective retention time is used to identify the collected compounds.</p>
<p><strong>Execution</strong>:<br>
For the VFA measurement, the filtrates (already diluted 1:10) that were prepared in the course of the ammonium determination (3.4.4) can be used. The samples are pipetted directly onto a second (common) 96-well plate, whereby the respective position must also be noted exactly! As volatile fatty acids can usually outgas very quickly and the measurement would therefore be subject to a large error, the plate must always be stored at 4 °C until the final measurement. As soon as all groups have loaded their samples, the plate is firmly sealed with adhesive foil.</p>
<p><strong>Evaluation</strong>:<br>
The output of the HPLC is a so-called chromatogram, in which different peaks follow one another in series. Each peak represents a specific chemical compound and the area under the peak describes its quantity (calculation via integration). The exact quantity normally has to be determined using a standard measurement. In our case, however, this step is no longer necessary as the method is already established. We receive the data in the form of a PDF data sheet with concentrations in mmol L<sup>-1</sup>, which are then displayed graphically over the test period.</p>
</section>
<section id="chemical-oxygen-demand-cod" class="level3">
<h3 class="anchored" data-anchor-id="chemical-oxygen-demand-cod">3.4.6. Chemical oxygen demand (COD)</h3>
<p>The chemical oxygen demand is an important parameter in wastewater treatment and biogas production and a measure of the sum of all substances present in the solution that can be oxidized (by dichromate). It can be used to monitor the degree of degradation in the reactor over time and also to estimate the theoretical biogas potential. It is assumed that a complete degradation of 1 g COD produces approx. 350 mL of methane (Reinhold, 2004). The total COD content is made up of the particulate and dissolved COD. The COD can be determined either by titration or photometrically. In the course of the practical course, photometric rapid kits are used to determine the total COD. Incubation and measurement are carried out in devices designed for this purpose (NANOCOLOR thermoblock and photometer; <b>Fig. 6</b>).</p>

<b>Fig. 6:</b> (<strong>A</strong>) <i>NANOCOLOR thermoblock for incubating the round cells and</i> (<strong>B</strong>) <i>NANOCOLOR photometer for measuring the round cells for COD determination in the reactor sludge.</i>

<p><strong>Execution</strong>:<br>
The thawed samples are first diluted 1:10 with A.d.. Pipette 2 mL of the solution into NANOCOLOR round cells. The cuvettes are then shaken (approx. 5 s) and incubated for 30 min at 160 °C in the NANOCOLOR heating block. After another shaking and an approx. half-hour cooling phase to room temperature, the cuvettes are measured with the NANOCOLOR photometer.</p>
<p><strong>Evaluation</strong>:<br>
The chemical oxygen demand for all preparations should be shown in the form of a bar chart in comparison to the control (mean value ± standard deviation). <strong>ATTENTION</strong>: The chemicals in the NANOCOLOR cuvettes are highly corrosive and harmful to health! For this reason, goggles and gloves must be worn when filling the cuvettes! After filling, the flasks must be carefully sealed. Both the incubation in the heating block and the subsequent cooling of the cuvettes must take place under the fume hood! The used cuvettes are put back into the box and then disposed of by the lab instructors.</p>
</section>
<section id="dry-matter-dm-and-organic-dry-matter-odm" class="level3">
<h3 class="anchored" data-anchor-id="dry-matter-dm-and-organic-dry-matter-odm">3.4.7. Dry matter (DM) and organic dry matter (oDM)</h3>
<p>As organic material usually contains different amounts of water, the dry matter is an important reference value and enables a direct comparison between different substrates. During anaerobic digestion, organic material is broken down and converted to a large extent into biogas. This leads to a reduction in (organic) dry matter, which is why TS and oTS are important parameters for monitoring biogas reactors. By continuously monitoring the values, the feeding strategy can be assessed and an excessively high or low rate can be detected. In addition, a sudden increase indicates a process malfunction and enables the operator to react in good time.</p>
<p><strong>Execution</strong>:<br>
Sludge samples (at least 30 g) from the pH measurement (3.4.1) or the EC measurement (3.4.2) are weighed into tared glass petri dishes (be sure to note the tare!) and the weight loss is determined at 105 °C overnight. Subsequently, approx. 1 g of the dried sample is weighed into ceramic dishes (also note the tare!) and incinerated for 5 hours at 550 °C in a muffle furnace (according to BGBl II 292/2001). The vessels must be cooled in a desiccator in the absence of oxygen both after drying at 105 °C and after muffling.</p>
<p><strong>Formulas for calculation</strong>:</p>
<p>DM [% FM] = Dried residue [g] x 100 / FM [g]<br>
WC [% FM] = 100% - DM [% FM]<br>
ioDM [% TS] = muffle residue [g] x 100 / DM [g]<br>
oDM [% TS] = 100% - ioDM [% TS]</p>
<p><strong>FM</strong> … <strong>F</strong>resh <strong>M</strong>atter<br>
<strong>WC</strong> … <strong>W</strong>ater <strong>C</strong>ontent<br>
<strong>ioDM</strong> … <strong>i</strong>n<strong>o</strong>rganic <strong>D</strong>ry <strong>M</strong>atter</p>
<p>oTS [% FM] = DM [% FM] x oDM [% TS]</p>
<p><strong>Evaluation</strong>:<br>
DM and oDM (expressed as a percentage of the fresh mass) should be shown graphically over time for all approaches and the control (mean ± standard deviation).</p>
</section>
<section id="carbonnitrogen-ratio-cn" class="level3">
<h3 class="anchored" data-anchor-id="carbonnitrogen-ratio-cn">3.4.8. Carbon/Nitrogen ratio (C/N)</h3>
<p>The ratio of carbon to nitrogen is of great importance in environmental microbiology, as bioavailable nitrogen is a limiting factor for many growth processes. On the other hand, the carbon contained is the basis for biogas (CH<sub>4</sub>, CO<sub>2</sub>) in anaerobic fermentation and therefore limits the entire process. For this reason, the C/N ratio during production should be within a certain range (approx. 20-30) (Yadvika et al., 2004). If the ratio is too low, ammonium can accumulate and the pH value in the reactor can rise, which is toxic for methanogens (Wang et al., 2012). The C/N ratio in the bioreactor can be adjusted and optimized by adding different co-substrates. The amount of carbon and nitrogen is determined during the practical course using elemental analysis (<b>Fig. 7</b>). This involves burning the samples at 850 °C and determining the ratio of the resulting gases (CO<sub>2</sub> and N<sub>2</sub>) using an infrared gas analyzer (IRGA).</p>
<p><strong>Execution</strong>:<br>
Samples from the TS determination (3.4.7) are ground to a fine powder and weighed into tin foils, if possible excluding residual air (max. 100-150 mg). The measurement is then carried out in the elemental analyzer. The tin foil must never be touched without gloves, as this could interfere with the measurement.</p>
<p><strong>Evaluation</strong>:<br>
The C/N ratio is calculated from the measured concentrations of carbon and nitrogen. The change in the C/N ratio over time should be shown graphically for all approaches compared to the control (mean ± standard deviation).</p>
</section>
</section>
<section id="hygienization-potential" class="level2">
<h2 class="anchored" data-anchor-id="hygienization-potential">3.5. Hygienization potential</h2>
<p>The use of fermentation residues as fertilizer has proven to be advantageous in many respects compared to non-fermented manure and slurry. One important reason for this is the hygienization of the substrate, which takes place in the course of anaerobic fermentation. This can increase the safety of spreading on meadows and fields. However, the efficiency of this hygienization depends on various parameters such as temperature and feeding strategy and can sometimes vary significantly. In the course of the practical course, we will determine the hygienization potential of the respective reactor systems using a classical, microbiological plating approach. Three different selective media (TBX Chromocult, XLT-4, PALCAM) will be used to detect relevant pathogens (coliform bacteria, <em>Salmonella</em> spp., <em>Listeria</em> spp.). Alternatively, pathogenic organisms could also be detected by molecular biology. Typical methods for this would be PCR (qualitative only), qPCR, dPCR, ddPCR or sequencing. In addition to the selective media, the total bacterial count (on standard medium) is also determined.</p>
<p><strong>Preparations</strong>:<br>
At the beginning of the practical course, sufficient plates are prepared for all analysis days and then stored in airtight packaging at 4 °C until use. The preparation is carried out according to the respective data sheets or <b>Tab. 3</b>.</p>

<b>Tab. 3:</b> <i>Composition of the standard medium for determining the total bacterial count.</i>

<table class="table">
<thead>
<tr class="header">
<th>Quantity</th>
<th>Ingredient</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>5.0 g</td>
<td>Peptone from meat</td>
</tr>
<tr class="even">
<td>2.5 g</td>
<td>Yeast extract</td>
</tr>
<tr class="odd">
<td>1.0 g</td>
<td>Glucose</td>
</tr>
<tr class="even">
<td>15.0 g</td>
<td>Agar</td>
</tr>
<tr class="odd">
<td>1000 mL</td>
<td>Fill up to 1000 mL with deionized water</td>
</tr>
<tr class="even">
<td>…adjust to pH = 7 (25 °C)</td>
<td></td>
</tr>
</tbody>
</table>
<p><strong>Execution</strong>:<br>
Taken reactor samples are diluted 1:10 with sterile isotonic NaCl solution (0.9% w/v) and shaken for 15 min at 200 rpm. After a sedimentation phase of 30 min, dilutions are prepared in steps of 10 (100, 10<sup>-1</sup>, 10<sup>-2</sup> for selective media; 10<sup>-5</sup>, 10<sup>-6</sup>, 10<sup>-7</sup> for standard medium) and approx. 25 µL each are plated on agar plates. Sterile working is essential here! The plates are incubated at 37 °C for 24 h and then analyzed. Only the concentration with the best countable number of CFUs is used. If no CFUs can be detected even at the lowest dilution, the plates are incubated further and evaluated again within the next few days. After counting, morphologically interesting colonies are transferred to a master plate with standard medium. Using a sterile pipette tip, some cell material is “picked” from free-standing, clearly demarcated individual colonies. To do this, the tip of the pipette only needs to touch the colony lightly to pick up some of it (the pipette itself or the suction function of the pipette is not necessary for this). The tip is then dabbed into an empty box on the master plate using the grid provided for orientation without damaging the agar. The master plate is then incubated at 37 °C until the picked individual colonies are clearly visible. In the laboratory notebook, the numbered boxes must be used to note which colony was picked from which plate (standard medium, TBX Chromocult, XLT-4, PALCAM) so that subsequent identification is possible.</p>
<p><strong>Evaluation</strong>:<br>
The number of CFUs should be shown graphically for both measurement days and compared with the control (mean ± standard deviation). In addition, the reduction of general germs or pathogens after nine days of running time should be indicated as a percentage. <strong>ATTENTION</strong>: The supplements used to prepare the XLT-4 and PALCAM agar are highly corrosive. Therefore, extreme caution must be exercised when preparing these media and protective goggles and suitable gloves must be worn! In addition, the agar plates must not be opened after cultivation, as they contain highly enriched pathogenic microorganisms (<strong>Exception</strong>: Colony PCR 3.6.1 under laminar flow)! For incubation and storage, the plates must be labeled with appropriate warning stickers and autoclaved immediately after evaluation or, in the case of the first week after colony PCR, immediately autoclaved.</p>
</section>
<section id="bio-molecular-characterization-of-pathogens" class="level2">
<h2 class="anchored" data-anchor-id="bio-molecular-characterization-of-pathogens">3.6. Bio-molecular characterization of pathogens</h2>
<section id="colony-pcr" class="level3">
<h3 class="anchored" data-anchor-id="colony-pcr">3.6.1. Colony PCR</h3>
<p>In general, specific DNA fragments are amplified in large numbers in the polymerase chain reaction (PCR), enabling even the smallest amounts of DNA to be analyzed. In contrast to conventional PCR, colony PCR does not use pure DNA as starting material. Instead, cell material (e.g.&nbsp;from a bacterial colony) is directly inoculated and the DNA it contains is then amplified. In this way, the time-consuming and relatively expensive step of DNA extraction can be avoided.</p>
<p><strong>Execution</strong>: Each group examines five morphologically interesting colonies from the master plates (3.5) of the first week. First, a master mix is prepared in sufficient quantity (number of samples examined + 1). This is done in sterile 0.5 mL Eppendorf tubes (do not forget to label!) according to the protocol in <b>Tab. 4</b>. After preparation, the tube must be tightly closed and vortexed for 5 s.</p>

<b>Tab. 4:</b> <i>Master mix for colony PCR analysis of selected pure cultures. Quantities for one sample.</i>

<table class="table">
<colgroup>
<col style="width: 25%">
<col style="width: 75%">
</colgroup>
<thead>
<tr class="header">
<th>Volume [µL]</th>
<th>Reagent</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>12.5</td>
<td>VWR Red Taq DNA Polymerase 2x Mastermix, 1.5 mM MgCl2</td>
</tr>
<tr class="even">
<td>0.5</td>
<td>Forward Primer: 27F (5’-AGA GTT TGA TCA TGG CTC A-3’)</td>
</tr>
<tr class="odd">
<td>0.5</td>
<td>Reverse primer: 1492R (5’-TAC GGT TAC CTT GTT ACG ACT T-3’)</td>
</tr>
<tr class="even">
<td>0.5</td>
<td>Bovine serum albumin (BSA, 2%)</td>
</tr>
<tr class="odd">
<td>11.0</td>
<td>PCR H<sub>2</sub>O</td>
</tr>
<tr class="even">
<td>25.0</td>
<td>TOTAL VOLUME</td>
</tr>
</tbody>
</table>
<p>Subsequently, 25 µL of master mix is pipetted into sterile PCR tubes and inoculated with cell material. For this purpose, suitable colonies are selected on the selective agar plates and small amounts are transferred to the tubes using sterile pipette tips. The finished tubes are vortexed for 5 s, briefly centrifuged with a tabletop centrifuge and placed in the thermal cycler (BioRad T100; <b>Fig. 9</b>). The program is started according to the instructions in <b>Tab. 5</b>. After completion, the tubes are stored at -20 °C until further use.</p>
<p><strong>ATTENTION</strong>: To over-inoculate the cell material, the agar plates with highly enriched bacteria must be opened. This may only be done under the laminar flow! The plates are only opened until enough cell material has been removed and are then closed again immediately. The pipette tip used and the laboratory gloves are disposed of in a specially labeled waste container. Hands must be disinfected after this step!</p>

<b>Tab. 5:</b> <i>Program for colony PCR testing of selected pathogens.</i>

<table class="table">
<thead>
<tr class="header">
<th>Phase</th>
<th>Temperature [°C]</th>
<th>Duration [s]</th>
<th>Cycles</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Initial denaturation</td>
<td>95</td>
<td>300</td>
<td>1</td>
</tr>
<tr class="even">
<td>Denaturation</td>
<td>95</td>
<td>30</td>
<td>35</td>
</tr>
<tr class="odd">
<td>Primer annealing</td>
<td>54</td>
<td>30</td>
<td>35</td>
</tr>
<tr class="even">
<td>Elongation</td>
<td>72</td>
<td>45</td>
<td>35</td>
</tr>
<tr class="odd">
<td>Final elongation</td>
<td>72</td>
<td>600</td>
<td>1</td>
</tr>
<tr class="even">
<td>End</td>
<td>12</td>
<td>-</td>
<td>-</td>
</tr>
</tbody>
</table>
</section>
<section id="dna-purification" class="level3">
<h3 class="anchored" data-anchor-id="dna-purification">3.6.2. DNA purification</h3>
<p>DNA amplified with the aid of PCR is usually subsequently purified (DNA clean-up) so that subsequent analyses are not disturbed by residues of the PCR reaction. In the practical course, purification is carried out using a special PCR clean-up kit (Sigma-Aldrich GenElute PCR clean-up). All centrifugation steps are carried out at 11,000 g. Procedure: Each sample is diluted 4:1 (e.g.&nbsp;25 µL PCR product + 75 µL buffer) with NT buffer in 2 mL Eppendorf tube (label!). Then NucleoSpin Extract Columns are inserted into fresh 2 mL Eppendorf tubes and loaded with the diluted samples. After centrifugation for 1 min (discard flow-through), 600 µL NT3 buffer is added and centrifuged again for 1 min. Now the flow-through is discarded and the tubes are centrifuged for another 2 min to dry the filter. The filters are placed in new, sterile 1.5 mL tubes and covered with 30 µL PCR water. After a short incubation phase (1 min at room temperature), centrifuge for 1 min (the filters are discarded) and store the eluate at 4 °C in the refrigerator until further use.</p>
</section>
<section id="gel-electrophoresis" class="level3">
<h3 class="anchored" data-anchor-id="gel-electrophoresis">3.6.3. Gel electrophoresis</h3>
<p>Gel electrophoresis is a common method for testing PCR samples. In the practical course, the analysis is carried out on a 1% agarose gel. The DNA fragments separate according to their length and are then visualized under UV light. A successful colony PCR should result in one clear band per sample.</p>
<p><strong>Preparations</strong>:<br>
To prepare the gel (one gel per group), 150 mg agarose and 15 mL 0.5x TAE buffer are filled into a 100 mL Erlenmeyer flask. The mixture is boiled in the microwave until the agarose has completely dissolved. Then add 1 µL DNA dye (SYBR Safe DNA gel stain), shake thoroughly and pour the liquid into a small gel chamber. Combs (optionally one or two) with a sufficient number of teeth are inserted into the still liquid gel. If necessary, air bubbles are removed with a sterile pipette tip. The gel takes about 15 minutes to harden completely and can be loaded.</p>
<p><strong>Execution</strong>:<br>
After removing the combs, the hardened gel including the holder is removed from the gel chamber and carefully placed in the electrophoresis chamber (<b>Fig. 10</b>). The liquid in the chamber must completely cover the gel, otherwise new 0.5x TAE buffer must be added. To load the gel, pipette 1.5 µL of sample into the pockets. The sequence must be noted! Pipette 1.5 µL DNA Ladder (GeneRuler Express) into a pocket (usually the first or last). Once all samples and the ladder have been loaded, the lid of the electrophoresis chamber is closed and the program is started at 100 V for 15 min. At the end of the running time, the gel is removed, carefully pushed onto the UV table and examined with the lid closed. The result can be photographed.</p>
<p><strong>ATTENTION</strong>: The marked safety zone must be observed! Objects standing there must not leave the area and separate gloves must be used (and then disposed of in the hazardous waste garbage can). The electrophoresis chamber must be unplugged when handling, otherwise there is a risk of electric shock. The device may only be plugged in and started when the lid is closed. The gel may only be examined on the UV table when the lid is closed. Direct UV light can otherwise cause damage, especially to the eyes!</p>
</section>
<section id="dna-quantification" class="level3">
<h3 class="anchored" data-anchor-id="dna-quantification">3.6.4. DNA quantification</h3>
<p>There are a number of different methods for determining the DNA content in a sample. In the practical course, quantification is carried out using a fluorometer (Promega Quantus). Alternatives would be the Nanodrop system or a photometric measurement with PicoGreen DNA dye.</p>
<p><strong>Execution</strong>:<br>
Each group prepares a stock solution for all their samples (+ 1). For each sample, 197.5 µL 1x TE buffer and 0.5 µL dye must be calculated. Then pipette 198 µL of the stock solution and 2 µL of sample into analysis tubes, vortex and measure in the Quantus Fluorometer (<b>Fig. 11</b>) after an incubation time of 2 min. The dye or all solutions containing the dye must be protected from light as much as possible (use aluminum foil!), as the substance is light-sensitive and can be destroyed quickly! If the measured values are above the measuring range, the affected samples must be diluted 1:10 with 1x TE buffer and measured again (the entire procedure must be repeated!).</p>
</section>
<section id="dna-sequencing-and-blasting" class="level3">
<h3 class="anchored" data-anchor-id="dna-sequencing-and-blasting">3.6.5. DNA sequencing and BLASTing</h3>
<p>There are different techniques for sequencing DNA. New systems such as Illumina, IonTorrent or Nanopore can analyze large quantities of sequences simultaneously and thus enable the analysis of entire genomes or complex environmental samples. Sanger sequencing is an older technology, but is perfectly suited to the sequencing of a pure culture due to its very low price and low susceptibility to errors. The samples in the practical course are therefore analyzed using Sanger sequencing and the DNA sequences obtained are then blasted. Blasting is the process of comparing a DNA sequence with an existing database and searching for the best matching entries. This enables the (taxonomic) assignment of an unknown DNA sequence. The quality of the database is decisive for the result! Common databases for bacterial gene fragments (usually 16S) are NCBI, Silva, Greengenes or RDP.</p>
<p><strong>Execution</strong>:<br>
Pipette 15 µL of the purified DNA extract from each sample into sterile 1.5 mL Eppendorf tubes. In addition, 2 µL of the forward primer (27F) used is added and the tubes are carefully sealed. Each tube is labeled with a prepaid sticker provided by the sequencing company. The code on the sticker and the corresponding sample name must then be entered into the online order form. The finished tubes are packed in an air bubble envelope and must be dropped into the company’s mailbox (1st floor, opposite the soil laboratory) by 4 pm at the latest. Within 24 hours, the results will be provided in the form of .fasta files. The DNA sequences contained therein are then analyzed using the online blast tool (https://blast.ncbi.nlm.nih.gov/Blast.cgi). We use the “Nucleotide collection” database from NCBI as a reference database.</p>
</section>
</section>
</section>
<section id="bibliography" class="level1">
<h1>4. Bibliography</h1>
<ol type="1">
<li>Ahring, B.K., Angelidaki, I., de Macario, C.C., Gavala, H., Hofman-Bang, J., Elfering, S.O., Raskin, L., Stams, A., Westermann, P., Zheng, D. 2003. Biomethanation I. Springer.<br>
</li>
<li>BMVIT. 2014. Zusammensetzung von Biogas. Available at: http://www.biogas-netzeinspeisung.at/technische-planung/funktionsprinzip-einer-biogasanlage/zusammensetzung-von-rohbiogas.html. Accessed: 22.06.2018<br>
</li>
<li>Boe, K., 2006. Online monitoring and control of the biogas process. Institute of Environment and Resources, Technical University of Denmark.<br>
</li>
<li>Conrad, R., Schink, B., Phelps, T.J., 1986. Thermodynamics of H<sub>2</sub>-consuming and H<sub>2</sub>-producing metabolic reactions in diverse methanogenic environments under in situ conditions. FEMS microbiology letters. 38(6), 353-360.<br>
</li>
<li>Drosg, B. 2013. Process monitoring in biogas plants. IEA Bioenergy Task.<br>
</li>
<li>EU, 2014. State of play on the sustainability of solid and gaseous biomass used for electricity, heating and cooling in the eu. On the WWW. URL https://ec.europa.eu/energy/sites/ener/files/2014_biomass_state_of_play_.pdf.<br>
</li>
<li>Goberna, M., Podmirseg, S., Waldhuber, S., Knapp, B., García, C., Insam, H., 2011. Pathogenic bacteria and mineral N in soils following the land spreading of biogas digestates and fresh manure. Applied soil ecology. 49, 18-25.<br>
</li>
<li>Hasegawa, S., Shiota, N., Katsura, K., Akashi, A., 2000. Solubilization of organic sludge by thermophilic aerobic bacteria as a pretreatment for anaerobic digestion. Water science and technology. 41(3), 163-169.<br>
</li>
<li>Hupfauf, S., Plattner, P., Wagner, A.O., Kaufmann, R., Insam, H., Podmirseg, S.M., 2018. Temperature shapes the microbiota in anaerobic digestion and drives efficiency to a maximum at 45° C. Bioresource technology. 269, 309-318.<br>
</li>
<li>Insam, H., Gómez-Brandón, M., Ascher, J., 2015. Manure-based biogas fermentation residues–Friend or foe of soil fertility? Soil Biology and Biochemistry. 84, 1-14.<br>
</li>
<li>Jain, S., Jain, S., Wolf, I.T., Lee, J., Tong, Y.W., 2015. A comprehensive review on operating parameters and different pretreatment methodologies for anaerobic digestion of municipal solid waste. Renewable and Sustainable Energy Reviews. 52, 142-154.<br>
</li>
<li>Kaiser, F.L. 2007. Einfluss der stofflichen Zusammensetzung auf die Verdaulichkeit nachwachsender Rohstoffe beim anaeroben Abbau in Biogasreaktoren, Technische Universität München.<br>
</li>
<li>Koster, I., Rinzema, A., De Vegt, A., Lettinga, G., 1986. Sulfide inhibition of the methanogenic activity of granular sludge at various pH-levels. Water Research. 20(12), 1561-1567.<br>
</li>
<li>Laaber, M., 2012. Gütesiegel Biogas: Evaluierung der technischen, ökologischen und sozioökonomischen Rahmenbedingungen für eine Ökostromproduktion aus Biogas. Universität für Bodenkultur Wien, Wien.<br>
</li>
<li>Labatut, R.A., Angenent, L.T., Scott, N.R., 2014. Conventional mesophilic vs.&nbsp;thermophilic anaerobic digestion: a trade-off between performance and stability? Water research. 53, 249-258.<br>
</li>
<li>Ma, J., Amjad Bashir, M., Pan, J., Qiu, L., Liu, H., Zhai, L., Rehim, A., 2018. Enhancing performance and stability of anaerobic digestion of chicken manure using thermally modified bentonite. Journal of Cleaner Production. 183, 11-19.<br>
</li>
<li>Meseguer-Lloret, S., Molins-Legua, C., Campins-Falco, P., 2002. Ammonium determination in water samples by using OPA-NAC reagent: a comparative study with nessler and ammonium selective electrode methods. International Journal of Environmental &amp; Analytical Chemistry. 82(7), 475-489.<br>
</li>
<li>Reinhold, F., 2004. TOC statt CSB – oder “New CSB”? Korrespondenz Abwasser. 51(7), 705.<br>
</li>
<li>Richards, B.K., Cummings, R.J., White, T.E., Jewell, W.J., 1991. Methods for kinetic analysis of methane fermentation in high solids biomass digesters. Biomass and Bioenergy. 1(2), 65-73.<br>
</li>
<li>Sahlström, L., 2003. A review of survival of pathogenic bacteria in organic waste used in biogas plants. Bioresource technology. 87(2), 161-166.<br>
</li>
<li>Taherzadeh, M.J., Karimi, K., 2008. Pretreatment of lignocellulosic wastes to improve ethanol and biogas production: a review. International journal of molecular sciences. 9(9), 1621-1651.<br>
</li>
<li>Tian, G., Yang, B., Dong, M., Zhu, R., Yin, F., Zhao, X., Wang, Y., Xiao, W., Wang, Q., Zhang, W., 2018. The effect of temperature on the microbial communities of peak biogas production in batch biogas reactors. Renewable Energy. 123, 15-25.<br>
</li>
<li>Wang, X., Yang, G., Feng, Y., Ren, G., Han, X., 2012. Optimizing feeding composition and carbon–nitrogen ratios for improved methane yield during anaerobic co-digestion of dairy, chicken manure and wheat straw. Bioresource Technology. 120, 78-83.<br>
</li>
<li>Weiland, P., 2008. Wichtige Messdaten für den Prozessablauf und Stand der Technik in der Praxis. Gülzower Fachgespräche. 27, 17-31.<br>
</li>
<li>Yadvika, Sreekrishnan, T., Kohli, S., Rana, V., 2004. Enhancement of biogas production from solid substrates using different techniques—-a review. Bioresource technology. 95(1), 1-10.</li>
</ol>
</section>
<section id="appendix" class="level1">
<h1>Appendix</h1>

<b>Tab. 6:</b> <i>Ampelsystem der BOKU Wien zur Einschätzung der gemessenen Parameter bezüglich der Arbeitsweise des Biogas-Reaktors (Laaber, 2012).</i>

<table class="table">
<colgroup>
<col style="width: 18%">
<col style="width: 14%">
<col style="width: 14%">
<col style="width: 29%">
<col style="width: 21%">
</colgroup>
<thead>
<tr class="header">
<th>Parameter</th>
<th>Einheit</th>
<th>Grün</th>
<th>Gelb</th>
<th>Rot</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>CSB</td>
<td>[g/kg]</td>
<td>40 - 90</td>
<td>&lt; 40; 90 - 110</td>
<td>&gt; 110</td>
</tr>
<tr class="even">
<td>TS</td>
<td>[% FM]</td>
<td>4 - 8</td>
<td>&lt; 4; 8 - 9</td>
<td>&gt; 9</td>
</tr>
<tr class="odd">
<td>oTS</td>
<td>[% FM]</td>
<td>3 - 6</td>
<td>&lt; 3; 6 - 7</td>
<td>&gt; 7</td>
</tr>
<tr class="even">
<td>NH<sub>4</sub>+</td>
<td>[g/kg]</td>
<td>&lt; 5</td>
<td>&gt; 5</td>
<td>-</td>
</tr>
<tr class="odd">
<td>Acetat</td>
<td>[g/L]</td>
<td>&lt; 1</td>
<td>1 - 3</td>
<td>&gt; 3</td>
</tr>
<tr class="even">
<td>Propionat</td>
<td>[mg/L]</td>
<td>&lt; 250</td>
<td>250 - 1000</td>
<td>&gt; 1000</td>
</tr>
<tr class="odd">
<td>Butyrat</td>
<td>[mg/L]</td>
<td>&lt; 50</td>
<td>50 - 100</td>
<td>&gt; 100</td>
</tr>
<tr class="even">
<td>Iso-Butyrat</td>
<td>[mg/L]</td>
<td>&lt; 60</td>
<td>60 - 200</td>
<td>&gt; 200</td>
</tr>
<tr class="odd">
<td>Valeriat</td>
<td>[mg/L]</td>
<td>&lt; 50</td>
<td>50 - 100</td>
<td>&gt; 100</td>
</tr>
<tr class="even">
<td>Iso-Valeriat</td>
<td>[mg/L]</td>
<td>&lt; 50</td>
<td>50 - 100</td>
<td>&gt; 100</td>
</tr>
<tr class="odd">
<td>VFA gesamt</td>
<td>[g/L]</td>
<td>&lt; 1.5</td>
<td>1.5 - 4.5</td>
<td>&gt; 4.5</td>
</tr>
<tr class="even">
<td>EC</td>
<td>[mS/cm]</td>
<td>18 - 20</td>
<td>&lt; 18; 20 - 60</td>
<td>&gt; 60</td>
</tr>
<tr class="odd">
<td>pH</td>
<td>-</td>
<td>7.5 - 8.1</td>
<td>7.1 - 7.5 &lt; 7.1;</td>
<td>&gt; 8.1</td>
</tr>
<tr class="even">
<td>FOS/TAC</td>
<td>-</td>
<td>0.3 - 0.5</td>
<td>0.2 - 0.3; 0.5 - 0.6</td>
<td>&lt; 0.2; &gt; 0.6</td>
</tr>
<tr class="odd">
<td>C/N</td>
<td>-</td>
<td>20 - 30</td>
<td>15 - 20; 30 - 35</td>
<td>&lt; 15; &gt; 35</td>
</tr>
</tbody>
</table>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>